pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -146,229 +169,174 @@ func (c *EtcdCertSignerController) sync(ctx context.Context, syncCtx factory.Syn
 			Reason: "AsExpected",
 		}))
 	return updateErr
-
 }
 
-func (c *EtcdCertSignerController) syncAllMasters(ctx context.Context, recorder events.Recorder) error {
-	certs, err := c.ensureCerts(ctx, recorder)
+func (c *EtcdCertSignerController) syncAllMasterCertificates(ctx context.Context, recorder events.Recorder) error {
+	// TODO(thomas): it is of utmost importance to keep the existing signer certs for now
+	// when we just create a new signer cert, the new revision does not allow the peer to join the existing two-node
+	// cluster based on the old CA. Any newly rotated additions will come for free through the signerCaBundle and will work out of the box.
+	// Unfortunately, we can't make use of the certrotation.RotatedSigningCASecret here because it would immediately try to rotate the existing signer.
+	signerCaPair, err := tlshelpers.ReadConfigSignerCert(ctx, c.secretClient)
 	if err != nil {
 		return err
 	}
 
-	// Write a secret that aggregates all certs for all nodes for the static
-	// pod controller to watch. A single secret ensures that a cert change
-	// (e.g. node addition or cert rotation) triggers at most one static pod
-	// rollout. If multiple secrets were written, the static pod controller
-	// might initiate rollout before all secrets had been updated.
-	secret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: operatorclient.TargetNamespace,
-			Name:      tlshelpers.EtcdAllCertsSecretName,
-			Annotations: map[string]string{
-				apiannotations.OpenShiftComponent: "Etcd",
-			},
-		},
-		Type: corev1.SecretTypeOpaque,
-		Data: certs,
+	// EnsureConfigMapCABundle is stateful w.r.t to the secret it manages, so we can simply add it to the bundle before the new one
```

Comment:
> w.r.t to the secret it manages

Is `EnsureConfigMapCABundle` managing a secret as well (i.e `etcd-signer`)? Thought this was for adding the cert to the `etcd-ca-bundle` configmap.
Or is this a typo and it's supposed to be:
```suggestion
	// EnsureConfigMapCABundle is stateful w.r.t to the configmap it manages, so we can simply add it to the bundle before the new one
```

Reply:
good catch, no, it's just managing the secret's content in a configmap, I'll merge your suggestion with the next change :)

Reply:
applied